src/test/java/seedu/findvisor/logic/parser/RemarkCommandParserTest.java

```diff
@@ -0,0 +1,63 @@
+package seedu.findvisor.logic.parser;
+
+import static seedu.findvisor.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.findvisor.logic.commands.CommandTestUtil.REMARK;
+import static seedu.findvisor.logic.parser.CliSyntax.PREFIX_REMARK;
+import static seedu.findvisor.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.findvisor.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.findvisor.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.findvisor.commons.core.index.Index;
+import seedu.findvisor.logic.commands.RemarkCommand;
+import seedu.findvisor.model.person.Remark;
+
+public class RemarkCommandParserTest {
+    private static final String MESSAGE_INVALID_FORMAT =
+            String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemarkCommand.MESSAGE_USAGE);
+    private RemarkCommandParser parser = new RemarkCommandParser();
+
+    @Test
+    public void parse_missingParts_failure() {
+        // no index specified
+        assertParseFailure(parser, REMARK, MESSAGE_INVALID_FORMAT);
+
+        // no index and no field specified
+        assertParseFailure(parser, "", MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidPreamble_failure() {
+        // negative index
+        assertParseFailure(parser, "-5" + REMARK, MESSAGE_INVALID_FORMAT);
+
+        // zero index
+        assertParseFailure(parser, "0" + REMARK, MESSAGE_INVALID_FORMAT);
+
+        // invalid arguments being parsed as preamble
+        assertParseFailure(parser, "1 some random string", MESSAGE_INVALID_FORMAT);
+
+        // invalid prefix being parsed as preamble
```

Comment:
Should this comment just be invalid prefix? because the preamble is 1 and that's valid

Reply:
Sorry selected the wrong lines, but I'm referring to line 43 and 44

Reply:
I believe the prefix is part of the preamble which is why it is still invalid prefix parsed as preamble?

Reply:
Preamble is defined as
> text before the first valid prefix

https://github.com/AY2324S2-CS2103-F15-1/tp/blob/21d80aeb405a3ce99fb00fd2ef5d977cf4eb3f60/src/main/java/seedu/findvisor/logic/parser/ArgumentMultimap.java#L58-L63